### DISCUSSION TITLE
error in react-native (not expo)

### DISCUSSION THREAD

**Discussion by steadev** [TanStack/query#7423] [Q&A]
I got below error when using any function in @tanstack/react-query.

Anyone can help me.. plz..!

RN: 0.69.7 (not using expo)
@tanstack/react-query: 5.36.0

Error: While trying to resolve module `@tanstack/react-query` from file `{PROJECT_PATH}/src/screens/Ask/queries/ask-queries.ts`, the package `{PROJECT_PATH}/node_modules/@tanstack/react-query/package.json` was successfully found. However, this package itself specifies a `main` module field that could not be resolved (`{PROJECT_PATH}/node_modules/@tanstack/react-query/build/legacy/index.cjs`. Indeed, none of these files exist:

  * {PROJECT_PATH}/node_modules/@tanstack/react-query/build/legacy/index.cjs(.native|.ios.js|.native.js|.js|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.svg|.native.svg|.svg)
  *{PROJECT_PATH}/node_modules/@tanstack/react-query/build/legacy/index.cjs/index(.native|.ios.js|.native.js|.js|.ios.json|.native.json|.json|.ios.ts|.native.ts|.ts|.ios.tsx|.native.tsx|.tsx|.ios.svg|.native.svg|.svg)

**Comment by mwmcode** [comment]
@steadev did you figure out what the issue was here?

**Reply by mwmcode** [reply]
adding `'cjs'` to metro config `resolver.sourceExts` fixed it for me
https://stackoverflow.com/questions/60124435

**Reply by steadev** [reply]
Thank you for your answer. 

I didn't figure out this issue.. but after i upgraded RN version to 0.74.1, this error solved..
